auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -593,8 +659,14 @@ public boolean verifyEvidence(final Evidence evidence) throws RealmUnavailableEx
                 ElytronMessages.log.tracef("Trying to authenticate identity %s using FileSystemSecurityRealm",
                         (loadedIdentity != null) ? loadedIdentity.getName() : "null");
             }
-
-            List<Credential> credentials = loadCredentials();
+            List<Credential> credentials = null;
+            try {
+                credentials = loadCredentials();
+            } catch (RealmUnavailableException e) {
+                    if (e.getMessage().contains("Signature for the following identity is invalid")) {
```

Comment:
Just a minor thing but looks like the spacing in this block is off.

Reply:
Instead of checking for a specific string here, it would be good to modify this to check the cause of the exception instead.

Reply:
When I check the cause of the exception (`e.getCause()`), it results in `null`

Reply:
Wherever the signature is determined to be invalid, instead of just throwing a `RealmUnvailableException` there, you can wrap the underlying the exception with a `RealmUnavailableException`. See `RealmUnavailableException(final Throwable cause)`.